### PR TITLE
Restrict documents to teams with access

### PR DESCRIPTION
**Pull request by KyleMacPherson**
https://trello.com/c/vwAqXGdC/1215-restrict-correspondence-documents-and-generic-attachments-to-teams-added-to-cases-5

## Checklist:
- [ ] Have you documented your changes in the pull request description?
- [ ] Does the change present any security considerations?
- [ ] Is any gem functionality overloaded? Eg: Devise controller methods being overloaded.
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

### General testing (author)
- [ ] Test without JavaScript
- [ ] Test on small screen

### Accessibility testing (author)
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable CSS - does content make sense and appear in a logical order?

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looks all good as expected.

**Review by stdstaff — APPROVED** [review]

File: app/controllers/active_storage/blobs/proxy_controller.rb [hunk 1]
```diff
@@ -22,6 +26,38 @@ def show
 private
 
   def authorize_blob
-    redirect_to "/sign-in" unless user_signed_in?
+    return redirect_to "/sign-in" unless user_signed_in?
+
+    if related_investigation
+      if attachment_is_correspondence_or_non_image_investigation_attachment? && !InvestigationPolicy.new(current_user, related_investigation).view_protected_details?
```
Comment (slorek) [inline app/controllers/active_storage/blobs/proxy_controller.rb:32]: Suggestion: move the `InvestigationPolicy.new(current_user, related_investigation)` instantiation to a helper method as it is repeated later on.

File: app/controllers/active_storage/blobs/proxy_controller.rb [hunk 2]
```diff
@@ -22,6 +26,38 @@ def show
...
+        return redirect_to "/", flash: { warning: "Not authorized to view this attachment" }
```
Comment (slorek) [inline app/controllers/active_storage/blobs/proxy_controller.rb:33]: Suggestion: Use British English spelling and define in I18n file.

File: app/controllers/active_storage/blobs/proxy_controller.rb [hunk 3]
```diff
@@ -22,6 +26,38 @@ def show
...
+
+  def is_an_image?
+    attachment_categorizer.is_an_image?
+  end
+
+  def is_an_investigation_image?
+    non_activity_attachment.record_type == "Investigation" && is_an_image?
+  end
+
+  def attachment_is_correspondence_or_non_image_investigation_attachment?
```
Comment (slorek) [inline app/controllers/active_storage/blobs/proxy_controller.rb:60]: Suggestion: name the method with a higher level meaning. The implementation is clearer.

```suggestion
  def attachment_is_protected_type?
```

**Review by slorek — APPROVED** [review]
Good work, just some tidy up suggestions.

**Pull request merged**